source/credentials_provider_ecs.c

```diff
@@ -71,6 +75,25 @@ struct aws_credentials_provider_ecs_user_data {
     int error_code;
 };
 
+/* called in between retries. */
+static void s_ecs_user_data_reset_request_specific_data(struct aws_credentials_provider_ecs_user_data *user_data) {
```

Comment:
There's another similarly named pre-existing function in this file: `s_aws_credentials_provider_ecs_user_data_reset_response()`

What's the difference? Can they be combined? If not, put them next to each other with a comment explaining when you'd call one and not the other. 

If there's already some kind of reset function in this file ... did we miss some other flow that should be reporting record_success() (or failure)?

Reply:
Thanks, I have removed the other function since it was not needed. We will call the new function between retries to clear the request level data.